docs/architecture/go_components.md

```diff
@@ -0,0 +1,117 @@
+# ClearlyDefined Go Support Architecture (Service)
+
+This drafts an architecture document for adding support for Go components to ClearlyDefined.
+
+## TLDR;
+
+* Add a new "go" type to ClearlyDefined
+* Only support go modules as components (which were added in Go 1.11). [According to the Go documentation](https://blog.golang.org/using-go-modules), modules are a dependency management system that makes dependency version information explicit and easier to manage. There were other, more complicated ways of managing go dependencies prior to this, but we should only focus on modules in our first iteration. We can revisit other ways of managing go dependencies later if necessary.
+* Support "proxy.golang.org" as the first go component provider, add other providers later if needed.
```

Comment:
This is the closet thing I can find to a central go package registry - it won't contain all packages, but we can add other providers as needed.

Reply:
>it won't contain all packages
Is this the case? I thought it is essentially a pass-through proxy to any resolvable Go package. ie: if `go get` could get it then it can be got through the proxy.

This makes me wonder if "proxy.golang.org" should be the provider or just "-" since the proxy is just a means to an end and the Go provider is essentially the internet.

Reply:
According to https://proxy.golang.org/ 

"proxy.golang.org does not save all modules forever. There are a number of reasons for this, but one reason is if proxy.golang.org is not able to detect a suitable license. In this case, only a temporarily cached copy of the module will be made available, and may become unavailable if it is removed from the original source and becomes outdated. The checksums will still remain in the checksum database regardless of whether or not they have become unavailable in the mirror."

Reply:
Ah ok, I guess it is client-side in the go tool that grabs a package directly if it doesn't exist on the proxy. From the Go users point of view, it is transparent.

Reply:
I think this might need to be influenced by the crawler design, if you can use `go get` to get the package, you may wish to not use a "provider", but if you will only access the proxy directly, then it makes sense.

Reply:
I did some experimenting with accessing the proxy directly - these URLs work (and should fit our purposes)

https://proxy.golang.org/rsc.io/quote/@v/list
https://proxy.golang.org/rsc.io/quote/@v/v1.3.0.mod
https://proxy.golang.org/rsc.io/quote/@v/v1.3.0.info
https://proxy.golang.org/rsc.io/quote/@v/v1.3.0.zip